flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connectors-doris/src/main/java/com/ververica/cdc/connectors/doris/sink/DorisMetadataApplier.java

```diff
@@ -0,0 +1,196 @@
+            DorisDatabase.applySchemaChangeEvent(event);
+            //send schema change op to doris
+            if (event instanceof CreateTableEvent){
+                applyCreateTableEvent((CreateTableEvent) event);
+            } else if (event instanceof AddColumnEvent) {
+                applyAddColumnEvent((AddColumnEvent) event);
+            } else if (event instanceof DropColumnEvent) {
+                applyDropColumnEvent((DropColumnEvent) event);
+            } else if (event instanceof RenameColumnEvent) {
+                applyRenameColumnEvent((RenameColumnEvent) event);
+            } else if (event instanceof AlterColumnTypeEvent) {
+                LOG.warn("AlterColumnType is not supported, skip.");
+            }
+        }catch (Exception ex){
+            //Catch exceptions to avoid affecting the main process
+            LOG.error("Schema change error, skip {}", event);
+        }
+    }
+
+    private void applyCreateTableEvent(CreateTableEvent event) throws IOException, IllegalArgumentException {
+        Schema schema = event.getSchema();
+        TableId tableId = event.tableId();
+        TableSchema tableSchema = new TableSchema();
+        tableSchema.setTable(tableId.getTableName());
+        tableSchema.setDatabase(tableId.getSchemaName());
+        tableSchema.setFields(buildFields(schema));
+        tableSchema.setDistributeKeys(buildDistributeKeys(schema));
+
+        if(CollectionUtil.isNullOrEmpty(schema.primaryKeys())){
+            tableSchema.setModel(DataModel.DUPLICATE);
+        }else{
+            tableSchema.setKeys(schema.primaryKeys());
+            tableSchema.setModel(DataModel.UNIQUE);
+        }
+        schemaChangeManager.createTable(tableSchema);
+    }
+
+    private Map<String, FieldSchema> buildFields(Schema schema){
+        Map<String, FieldSchema> fieldSchemaMap = new HashMap<>();
+        List<String> columnNameList = schema.getColumnNames();
+        for(String columnName : columnNameList){
+            Column column = schema.getColumn(columnName).get();
+            String dorisTypeStr = convertToDorisType(column.getType());
+            fieldSchemaMap.put(column.getName(), new FieldSchema(column.getName(), dorisTypeStr, column.getComment()));
+        }
+        return fieldSchemaMap;
+    }
+
+    private List<String> buildDistributeKeys(Schema schema){
+        if(!CollectionUtil.isNullOrEmpty(schema.primaryKeys())){
+            return schema.primaryKeys();
+        }
+        if(!CollectionUtil.isNullOrEmpty(schema.getColumnNames())){
+            return Collections.singletonList(schema.getColumnNames().get(0));
+        }
+        return new ArrayList<>();
+    }
+
+    private void applyAddColumnEvent(AddColumnEvent event) throws IOException, IllegalArgumentException {
+        TableId tableId = event.tableId();
+        List<AddColumnEvent.ColumnWithPosition> addedColumns = event.getAddedColumns();
+        for(AddColumnEvent.ColumnWithPosition col: addedColumns){
```

Comment:
Yes, because adding a **value column** in Doris does not support adding it to the middle of multiple **Key columns**.
At the same time, the default is to use JSON format to import. The order of the columns will not affect the data quality. The value column can be directly appended to the end.